src/components/Reports/ReportEdit.tsx

```diff
@@ -56,6 +56,46 @@ export default function ReportEdit({ organization }: ReportEditProps): JSX.Eleme
     }
   }, [reportIdInt, snackbar]);
 
+  const [newReportFiles, setNewReportFiles] = useState<File[]>();
+
+  const [initialReportFiles, setInitialReportFiles] = useState<ReportFile[]>();
+
+  const [updatedReportFiles, setUpdatedReportFiles] = useState<ReportFile[]>();
+
+  const updateFiles = async () => {
+    await Promise.all(
+      initialReportFiles?.map((f) => {
+        if (!updatedReportFiles?.includes(f)) {
+          return ReportService.deleteReportFile(reportIdInt, f.id);
+        }
+        return undefined;
+      }) ?? []
+    );
+    await Promise.all(newReportFiles?.map((f) => ReportService.uploadReportFile(reportIdInt, f)) ?? []);
+  };
+
+  useEffect(() => {
+    const getFiles = async () => {
+      if (report) {
+        const fileListResponse = await ReportService.getReportFiles(report.id);
+        if (!fileListResponse.requestSucceeded || fileListResponse.error) {
```

Comment:
i think we can skip the || check since we won't have a success case with an error message